community/kernel/src/main/java/org/neo4j/kernel/api/cursor/PropertyCursor.java

```diff
@@ -29,6 +30,52 @@
 public interface PropertyCursor
         extends Cursor
 {
+    Function<PropertyCursor, DefinedProperty> GET_PROPERTY = new Function<PropertyCursor, DefinedProperty>()
+    {
+        @Override
+        public DefinedProperty apply( PropertyCursor propertyCursor )
+        {
+            return propertyCursor.getProperty();
+        }
+    };
+
+    ToIntFunction<PropertyCursor> GET_KEY_INDEX_ID = new ToIntFunction<PropertyCursor>()
+    {
+        @Override
+        public int apply( PropertyCursor value )
+        {
+            return value.getProperty().propertyKeyId();
+        }
+    };
+
+    PropertyCursor EMPTY = new PropertyCursor()
+    {
+        @Override
+        public boolean seek( int keyId )
+        {
+            return false;
+        }
+
+        @Override
+        public DefinedProperty getProperty()
+        {
+            return null;
```

Comment:
I don't know about the contract of PropertyCursor in general, but wouldn't it be better to throw, say an `IllegalStateException` instead of returning `null`? Also noting that in the javadoc, such that if `getProperty` and similar methods are called before a successful `seek` then that exception is thrown.